src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/CompactConstantPoolsPlugin.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2015, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.tools.jlink.internal.plugins;
+
+import java.util.Map;
+import java.util.function.Function;
+
+import jdk.tools.jlink.internal.ResourcePoolManager.ResourcePoolImpl;
+import jdk.tools.jlink.plugin.ResourcePool;
+import jdk.tools.jlink.plugin.ResourcePoolBuilder;
+import jdk.tools.jlink.internal.ImagePluginStack;
+import jdk.tools.jlink.internal.ResourcePoolManager;
+import jdk.tools.jlink.internal.ResourcePrevisitor;
+import jdk.tools.jlink.internal.StringTable;
+
+/**
+ *
+ * ZIP and String Sharing compression plugin
+ */
+public final class CompactConstantPoolsPlugin extends AbstractPlugin implements ResourcePrevisitor {
+
+    private static final String FILTER = "filter";
+    private StringSharingPlugin ss;
+
+    public CompactConstantPoolsPlugin() {
+        super("compact-constant-pools");
```

Comment:
This plugin needs a better option name.   Maybe `--share-utf8-entries`?

I also wonder if anyone really uses this plugin.  It writes the shared UTF8 strings into the jimage.  It incurs overhead in reconstructing the constant pool when loading classes.    So I wonder if we just drop this plugin - @AlanBateman, @igraves what do you think?

Reply:
If we are planning to drop this, we will not need to introduce a new plugin option but instead states that `--compress 1` will be deprecated and removed in a future release.

Reply:
That makes sense to me. May want @AlanBateman's opinion before final action.